src/bin/cratesfyi.rs

```diff
@@ -325,3 +36,402 @@ fn logger_init() {
+
+    /// Unlocks cratesfyi daemon to continue building new crates
+    Unlock,
+
+    PrintOptions,
+}
+
+impl BuildSubcommand {
+    pub fn handle_args(self, mut docbuilder: DocBuilder) {
+        match self {
+            Self::World => {
+                docbuilder.load_cache().expect("Failed to load cache");
+
+                let mut builder = RustwideBuilder::init().unwrap();
+                builder
+                    .build_world(&mut docbuilder)
+                    .expect("Failed to build world");
+
+                docbuilder.save_cache().expect("Failed to save cache");
+            }
+
+            Self::Crate {
+                crate_name,
+                crate_version,
+                local,
+            } => {
+                docbuilder.load_cache().expect("Failed to load cache");
+                let mut builder = RustwideBuilder::init().expect("failed to initialize rustwide");
+
+                if let Some(path) = local {
+                    builder
+                        .build_local_package(&mut docbuilder, &path)
+                        .expect("Building documentation failed");
+                } else {
+                    builder
+                        .build_package(&mut docbuilder, &crate_name, &crate_version, None)
+                        .expect("Building documentation failed");
+                }
+
+                docbuilder.save_cache().expect("Failed to save cache");
+            }
+
+            Self::UpdateToolchain { only_first_time } => {
+                if only_first_time {
+                    let conn = db::connect_db().unwrap();
+                    let res = conn
+                        .query("SELECT * FROM config WHERE name = 'rustc_version';", &[])
+                        .unwrap();
+
+                    if !res.is_empty() {
+                        println!("update-toolchain was already called in the past, exiting");
+                        return;
+                    }
+                }
+
+                let mut builder = RustwideBuilder::init().unwrap();
+                builder
+                    .update_toolchain()
+                    .expect("failed to update toolchain");
+            }
+
+            Self::AddEssentialFiles => {
+                let mut builder = RustwideBuilder::init().unwrap();
+                builder
+                    .add_essential_files()
+                    .expect("failed to add essential files");
+            }
+
+            Self::Lock => docbuilder.lock().expect("Failed to lock"),
+            Self::Unlock => docbuilder.unlock().expect("Failed to unlock"),
+            Self::PrintOptions => println!("{:?}", docbuilder.options()),
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, StructOpt)]
+enum DatabaseSubcommand {
+    #[structopt(name = "move-to-s3")]
+    MoveToS3,
+
+    /// Run database migrations
+    Migrate {
+        /// The database version to migrate to
+        #[structopt(name = "VERSION")]
+        version: i64,
```

Comment:
This version should be optional. If not present, it should migrate to the latest version.

Reply:
Easy enough to fix, I’ll do it tomorrow